File: Common/Entity/TaskPlannerService/TaskCategory.cs
using System.ComponentModel.DataAnnotations;

namespace Common.Entity.TaskPlannerService
{
    public class TaskCategory
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Color { get; set; }

        public string ImagePath { get; set; }

        [Required]
        public bool IsOn { get; set; }
    }
}

File: MyHomeSPA/src/app/tasks/models/taskCategory.ts
export class TaskCategory {

    public id: number;

    public name: string;

    public color: string;

    public imagePath: string;

    public isOn: boolean;
    
    constructor() {
        
        this.id = 0;

        this.name = "Category";

        this.color = '000';

        this.imagePath = 'https://st3.depositphotos.com/13821126/18366/v/1600/depositphotos_183663702-stock-illustration-sale-bags-packages-in-the.jpg';

        this.isOn = true;
    }
}